### DISCUSSION TITLE
Tensor networks in Qutip

### DISCUSSION THREAD

**Discussion by EmilianoG-byte** [qutip/qutip#2232] [General]
Hi!

I was reading this[ interesting post](https://agalicia.netlify.app/post/internship_1/) about the possible usefulness of tensor networks as a backend for Qutip simulations by @AGaliciaMartinez. It seems to be already a while since it was published, is there any plans to include this into Qutip?

Thank you!

**Comment by hodgestar** [comment]
Hi @EmilianoG-byte. You could look at https://github.com/qutip/qutip-tensornetwork and try it out. It hasn't been looked at since @AGaliciaMartinez finished work on it, so it might need a bit of updating, but on the plus side some of the new QuTiP solvers should be able to work with it.

**Reply by EmilianoG-byte** [reply]
I see. So, it means that there is no plans of creating a purely tensor-network-based solver for Qutip? E.g. using MPS and MPO's for time evolution.

**Reply by hodgestar** [reply]
The custom QuTiP solvers will already use only the tensor network representation of the RHS. They won't convert the RHS to a full sparse or dense matrix.

It is also possible to add other solvers specifically for the tensor network representation but none exist yet.

**Reply by EmilianoG-byte** [reply]
So solvers like `mesolve` and `mcsolve` already use the tensor representation under the hood? Or is this only coming with the 5.0 release?

**Reply by hodgestar** [reply]
Only with 5.0 releases (alpha releases are available for trying out). If you are interested, please give it a try and ask if you run into issues.

**Reply by EmilianoG-byte** [reply]
Sounds great! I will discuss it with my colleagues since I guess we would need to make some changes on our code when migrating to the newest QuTip version. Thank you for your help!

**Reply by hodgestar** [reply]
Pleasure.

If your code doesn't reach too deep into QuTiP's internals, migrating to QuTiP 5 shouldn't be too painful. If you do find something that is hard to migrate, let us know.
